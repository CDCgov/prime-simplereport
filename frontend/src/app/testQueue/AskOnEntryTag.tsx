import { useState, useEffect } from "react";

import { AoEAnswers } from "./AoEForm/AoEForm";

export const areAnswersComplete = (answers: AoEAnswers) => {
  if (!answers.noSymptoms) {
    const symptoms = JSON.parse(answers.symptoms);
    const filled = Object.values(symptoms).some((v) => String(v) === "true");
    if (!filled || !answers.symptomOnset) {
      return false;
    }
  }
  if (!answers.pregnancy) {
    return false;
  }
  return true;
};

interface Props {
  aoeAnswers: AoEAnswers;
}

const AskOnEntryTag: React.FC<Props> = ({ aoeAnswers }) => {
  const [answers, setAnswers] = useState(aoeAnswers);
  useEffect(() => {
    setAnswers(aoeAnswers);
  }, [aoeAnswers]);
  if (areAnswersComplete(answers)) {
    return <span className="usa-tag bg-green">COMPLETED</span>;
  } else {
    return <span className="usa-tag">PENDING</span>;
  }
};

export default AskOnEntryTag;
