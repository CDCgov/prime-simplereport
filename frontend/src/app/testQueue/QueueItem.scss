.prime-card-container {
  .card-header {
    display: flex;
    justify-content: flex-start;
    align-items: center;

    .card-name {
      font-size: 22px;
      font-weight: 700;
      color: #005ea2;
      cursor: pointer;
    }

    .card-dob {
      margin-left: 24px;
      font-size: 16px;
      font-weight: 400;
      color: #71767a;

      .card-date {
        font-weight: 700;
      }
    }
  }

  .card-test-input {
    height: 34px;
    margin-right: 12px;
    max-width: 130px;
    animation: 0.25s fadeIn;
    animation-fill-mode: both;

    @keyframes fadeIn {
      0% {
        opacity: 0;
      }

      100% {
        visibility: visible;
        opacity: 1;
      }
    }

    &__error {
      border-color: #d54309;
    }
  }

  .card-dropdown {
    select {
      height: 34px;
      padding-top: 4px;
      padding-bottom: 4px;
    }
  }

  input[type="date"]::-webkit-calendar-picker-indicator {
    margin-left: 0;
  }

  .check-box-container {
    display: flex;
    align-items: center;
    height: 34px;
  }

  .test-questionnaire-btn {
    font-weight: 700 !important;
    margin-bottom: 1em;
  }

  .flex-col-container {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    flex-wrap: wrap;
  }

  .test-date-time-container {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    flex-wrap: wrap;
    margin-top: 8px;
  }

  .queue-item-error-message {
    color: #d54309;
    display: block;
  }
}

.symptom-onset-date {
  height: 40px;
  width: 200px;
  margin-top: 8px;
}
