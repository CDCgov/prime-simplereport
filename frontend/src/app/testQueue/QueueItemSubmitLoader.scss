.sr-queue-item-submit-loader {
  background-color: rgba(255, 255, 255, 0.92);

  &-enter {
    opacity: 0;
  }

  &-enter-active {
    display: block;
    opacity: 1;
    transition: opacity 300ms;
  }
}
