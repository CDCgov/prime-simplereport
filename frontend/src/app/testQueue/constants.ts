import { AlertContent } from "../commonComponents/Alert";
import { displayFullName } from "../utils";

export const QUEUE_NOTIFICATION_TYPES = {
  ADDED_TO_QUEUE__SUCCESS: 1,
  SUBMITTED_RESULT__SUCCESS: 2,
  DELIVERED_RESULT_TO_PATIENT__FAILURE: 3,
};

export type SomeoneWithName = {
  firstName?: string | null;
  middleName?: string | null;
  lastName?: string | null;
};

export const ALERT_CONTENT = {
  [QUEUE_NOTIFICATION_TYPES.ADDED_TO_QUEUE__SUCCESS]: <
    T extends SomeoneWithName
  >(
    patient: T,
    startWithLastName: boolean = true
  ): AlertContent => {
    return {
      type: "success",
      title: `${displayFullName(
        patient.firstName,
        patient.middleName,
        patient.lastName,
        startWithLastName
      )} was added to the queue`,
      body: "Newly added patients go to the bottom of the queue",
    };
  },
  [QUEUE_NOTIFICATION_TYPES.SUBMITTED_RESULT__SUCCESS]: <
    T extends SomeoneWithName
  >(
    patient: T,
    startWithLastName: boolean = true,
    dataRetentionLimitsEnabled: boolean = false
  ): AlertContent => {
    const resultText = dataRetentionLimitsEnabled
      ? "has been sent."
      : "was saved and reported.";
    return {
      type: "success",
      title: `Result for ${displayFullName(
        patient.firstName,
        patient.middleName,
        patient.lastName,
        startWithLastName
      )} ${resultText}`,
      body: "See Results to view all test submissions",
    };
  },
  [QUEUE_NOTIFICATION_TYPES.DELIVERED_RESULT_TO_PATIENT__FAILURE]: <
    T extends SomeoneWithName
  >(
    patient: T,
    startWithLastName: boolean = true
  ): AlertContent => {
    return {
      type: "error",
      title: `Unable to text result to ${displayFullName(
        patient.firstName,
        patient.middleName,
        patient.lastName,
        startWithLastName
      )}`,
      body: "The phone number provided may not be valid or may not be able to accept text messages",
    };
  },
};

export const MIN_SEARCH_CHARACTER_COUNT = 2;

export const SEARCH_DEBOUNCE_TIME = 500;
