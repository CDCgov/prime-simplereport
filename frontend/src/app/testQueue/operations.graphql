
mutation RemovePatientFromQueue($patientId: ID!) {
    removePatientFromQueue(patientId: $patientId)
  }

mutation EditQueueItemMultiplex(
    $id: ID!
    $deviceId: String
    $deviceSpecimenType: ID
    $results: [MultiplexResultInput]
    $dateTested: DateTime
  ) {
    editQueueItemMultiplex(
      id: $id
      deviceId: $deviceId
      deviceSpecimenType: $deviceSpecimenType
      results: $results
      dateTested: $dateTested
    ) {
      results {
        disease {
          name
        }
        testResult
      }
      dateTested
      deviceType {
        internalId
        testLength
      }
      deviceSpecimenType {
        internalId
        deviceType {
          internalId
          testLength
        }
        specimenType {
          internalId
        }
      }
    }
  }

mutation SubmitTestResultMultiplex(
    $patientId: ID!
    $deviceId: String!
    $deviceSpecimenType: ID
    $results: [MultiplexResultInput]!
    $dateTested: DateTime
  ) {
    addTestResultMultiplex(
      patientId: $patientId
      deviceId: $deviceId
      deviceSpecimenType: $deviceSpecimenType
      results: $results
      dateTested: $dateTested
    ) {
      testResult {
        internalId
      }
      deliverySuccess
    }
  }
