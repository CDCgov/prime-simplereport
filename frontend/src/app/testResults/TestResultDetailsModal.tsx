import { gql } from "@apollo/client";
import Modal from "react-modal";
import classnames from "classnames";

import iconClose from "../../img/close.svg";
import "./TestResultPrintModal.scss";
import { QueryWrapper } from "../commonComponents/QueryWrapper";
import { TestResult } from "../testQueue/QueueItem";
import { formatFullName } from "../utils/user";
import { symptomsStringToArray } from "../utils/symptoms";
import {
  PregnancyCode,
  pregnancyMap,
} from "../../patientApp/timeOfTest/constants";
import { formatDateWithTimeOption } from "../utils/date";

type Result = {
  dateTested: string;
  result: TestResult;
  correctionStatus: TestCorrectionStatus;
  noSymptoms: boolean;
  symptoms: string;
  symptomOnset: string;
  pregnancy: PregnancyCode;
  deviceType: {
    name: string;
  };
  patient: {
    firstName: string;
    middleName: string;
    lastName: string;
    birthDate: string;
  };
  createdBy: {
    name: {
      firstName: string;
      middleName: string;
      lastName: string;
    };
  };
};

export const testResultDetailsQuery = gql`
  query getTestResultDetails($id: ID!) {
    testResult(id: $id) {
      dateTested
      result
      correctionStatus
      symptoms
      symptomOnset
      pregnancy
      deviceType {
        name
      }
      patient {
        firstName
        middleName
        lastName
        birthDate
      }
      createdBy {
        name {
          firstName
          middleName
          lastName
        }
      }
    }
  }
`;

const labelClasses = "text-bold text-no-strike text-ink";
const containerClasses =
  "width-full font-sans-md add-list-reset border-base-lighter border-2px radius-md padding-x-2 padding-y-1";
const strikeClasses = "text-base text-strike";

interface Props {
  data: { testResult: Nullable<Result> };
  testResultId: string;
  closeModal: () => void;
}

export const DetachedTestResultDetailsModal = ({ data, closeModal }: Props) => {
  const {
    dateTested,
    result,
    correctionStatus,
    symptoms,
    symptomOnset,
    pregnancy,
    deviceType,
    patient,
    createdBy,
  } = { ...data?.testResult };

  const removed = correctionStatus === "REMOVED";
  const symptomList = symptoms ? symptomsStringToArray(symptoms) : [];

  return (
    <Modal
      isOpen={true}
      style={{
        content: {
          maxHeight: "90vh",
          width: "50em",
          position: "initial",
        },
      }}
      overlayClassName="prime-modal-overlay display-flex flex-align-center flex-justify-center"
      contentLabel="Unsaved changes to current user"
      ariaHideApp={process.env.NODE_ENV !== "test"}
    >
      <div className="display-flex flex-justify">
        <h1 className="font-heading-lg margin-top-05 margin-bottom-0">
          Test details
        </h1>
        <div className="sr-time-of-test-buttons">
          <button
            className="modal__close-button"
            style={{ cursor: "pointer" }}
            onClick={closeModal}
          >
            <img className="modal__close-img" src={iconClose} alt="Close" />
          </button>
        </div>
      </div>
      <div className="border-top border-base-lighter margin-x-neg-205 margin-top-1"></div>
      <h2 className="font-sans-md margin-top-3">Patient</h2>
      <div
        className={classnames(containerClasses, "grid-row flex-row padding-0")}
      >
        <div className="grid-col padding-2 border-right-1px border-base-lighter">
          <span
            className={classnames(
              labelClasses,
              "display-block margin-bottom-1"
            )}
          >
            Name
          </span>
          <span
            className={classnames("font-sans-lg", removed && strikeClasses)}
          >
            {patient ? formatFullName(patient) : "--"}
          </span>
        </div>
        <div className="grid-col padding-2">
          <span
            className={classnames(
              labelClasses,
              "display-block margin-bottom-1"
            )}
          >
            Date of birth
          </span>
          <span
            className={classnames("font-sans-lg", removed && strikeClasses)}
          >
            {patient?.birthDate
              ? formatDateWithTimeOption(patient.birthDate)
              : "--"}
          </span>
        </div>
      </div>
      <h2 className="font-sans-md margin-top-3">Test details</h2>
      <table className={containerClasses}>
        <tbody>
          <DetailsRow
            label="SARS-CoV-2 result"
            value={result}
            removed={removed}
          />
          <DetailsRow
            label="Test date"
            value={dateTested && formatDateWithTimeOption(dateTested, true)}
            removed={removed}
          />
          <DetailsRow
            label="Device"
            value={deviceType && deviceType.name}
            removed={removed}
          />
          <DetailsRow
            label="Symptoms"
            value={
              symptomList.length > 0 ? symptomList.join(", ") : "No symptoms"
            }
            removed={removed}
          />
          <DetailsRow
            label="Symptom onset"
            value={symptomOnset && formatDateWithTimeOption(symptomOnset)}
            indent
            removed={removed}
          />
          <DetailsRow
            label="Pregnant?"
            value={pregnancy && pregnancyMap[pregnancy]}
            removed={removed}
          />
          <DetailsRow
            label="Submitted by"
            value={createdBy?.name && formatFullName(createdBy.name)}
            removed={removed}
            last
          />
        </tbody>
      </table>
    </Modal>
  );
};

const TestResultDetailsModal = (props: Omit<Props, "data">) => (
  <QueryWrapper<Props>
    query={testResultDetailsQuery}
    queryOptions={{ variables: { id: props.testResultId } }}
    Component={DetachedTestResultDetailsModal}
    componentProps={{ ...props }}
    displayLoadingIndicator={false}
  />
);

export default TestResultDetailsModal;

type StringOrFalsey = string | null | undefined | false;

const DetailsRow = ({
  label,
  value,
  indent,
  last,
  removed,
}: {
  label: string;
  value: StringOrFalsey;
  indent?: boolean;
  last?: boolean;
  removed?: boolean;
}) => {
  const tdClasses = classnames(
    "padding-y-3",
    !last && "border-bottom-1px border-base-lighter"
  );

  const labelColumnClasses = classnames(
    labelClasses,
    tdClasses,
    "width-card-lg text-left",
    indent && "font-sans-sm padding-left-2"
  );

  return (
    <tr>
      <th className={labelColumnClasses}>{label}</th>
      <td className={classnames(tdClasses, removed && strikeClasses)}>
        {value || "--"}
      </td>
    </tr>
  );
};
