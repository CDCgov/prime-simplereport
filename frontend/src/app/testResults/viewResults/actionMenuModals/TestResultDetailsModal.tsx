import { gql } from "@apollo/client";
import Modal from "react-modal";
import classnames from "classnames";

import iconClose from "../../../../img/close.svg";
import { QueryWrapper } from "../../../commonComponents/QueryWrapper";
import { formatFullName } from "../../../utils/user";
import { symptomsStringToArray } from "../../../utils/symptoms";
import { Result as ResponseResult } from "../../../../generated/graphql";
import {
  PregnancyCode,
  pregnancyMap,
} from "../../../../patientApp/timeOfTest/constants";
import {
  getResultByDiseaseName,
  hasMultiplexResults,
} from "../../../utils/testResults";
import { displayFullName } from "../../../utils";
import { formatDateWithTimeOption } from "../../../utils/date";
import { MULTIPLEX_DISEASES } from "../../constants";

import "./TestResultPrintModal.scss";

type Result = {
  dateTested: string;
  results: MultiplexResult[];
  correctionStatus: TestCorrectionStatus;
  noSymptoms: boolean;
  symptoms: string;
  symptomOnset: string;
  pregnancy: PregnancyCode;
  deviceType: {
    name: string;
  };
  patient: {
    firstName: string;
    middleName: string;
    lastName: string;
    birthDate: string;
  };
  createdBy: {
    name: {
      firstName: string;
      middleName: string;
      lastName: string;
    };
  };
};

export const testResultDetailsQuery = gql`
  query getTestResultDetails($id: ID!) {
    testResult(id: $id) {
      dateTested
      results {
        disease {
          name
        }
        testResult
      }
      correctionStatus
      symptoms
      symptomOnset
      pregnancy
      deviceType {
        name
      }
      patient {
        firstName
        middleName
        lastName
        birthDate
      }
      createdBy {
        name {
          firstName
          middleName
          lastName
        }
      }
    }
  }
`;

const labelClasses = "text-bold text-no-strike text-ink";
const containerClasses =
  "width-full font-sans-md add-list-reset border-base-lighter border-2px radius-md padding-x-2 padding-y-1";
const strikeClasses = "text-base text-strike";

interface TestResultDetailsModalProps {
  isOpen: boolean;
  testResult: ResponseResult | undefined;
  closeModal: () => void;
}
interface DetachedTestResultDetailsModalProps {
  data: { testResult: Nullable<Result> };
  closeModal: () => void;
}

export const DetachedTestResultDetailsModal = ({
  data,
  closeModal,
}: DetachedTestResultDetailsModalProps) => {
  const dateTested = data?.testResult.dateTested;

  const results = data?.testResult.results;
  const correctionStatus = data?.testResult.correctionStatus;
  const symptoms = data?.testResult.symptoms;
  const symptomOnset = data?.testResult.symptomOnset;
  const pregnancy = data?.testResult.pregnancy;
  const deviceType = data?.testResult.deviceType;
  const patient = data?.testResult.patient;
  const createdBy = data?.testResult.createdBy;

  const removed = correctionStatus === "REMOVED";
  const symptomList = symptoms ? symptomsStringToArray(symptoms) : [];
  const displayResult: { [diseaseResult: string]: TestResult | null } = {
    covidResult: results
      ? getResultByDiseaseName(results, MULTIPLEX_DISEASES.COVID_19)
      : "UNKNOWN",
  };
  const multiplexEnabled = results && hasMultiplexResults(results);
  if (multiplexEnabled) {
    displayResult["fluAResult"] = getResultByDiseaseName(
      results,
      MULTIPLEX_DISEASES.FLU_A
    );
    displayResult["fluBResult"] = getResultByDiseaseName(
      results,
      MULTIPLEX_DISEASES.FLU_B
    );
  }
  return (
    <>
      <div className="display-flex flex-justify">
        <h1
          id="result-detail-title"
          className="font-heading-lg margin-top-05 margin-bottom-0"
        >
          Result details
        </h1>
        <div className="sr-time-of-test-buttons">
          <button
            className="modal__close-button"
            style={{ cursor: "pointer" }}
            onClick={closeModal}
          >
            <img className="modal__close-img" src={iconClose} alt="Close" />
          </button>
        </div>
      </div>
      <div className="border-top border-base-lighter margin-x-neg-205 margin-top-1"></div>
      <h2 className="font-sans-md margin-top-3">Patient</h2>
      <div
        className={classnames(containerClasses, "grid-row flex-row padding-0")}
      >
        <div className="grid-col padding-2 border-right-1px border-base-lighter">
          <span
            className={classnames(
              labelClasses,
              "display-block margin-bottom-1"
            )}
          >
            Name
          </span>
          <span
            className={classnames("font-sans-lg", removed && strikeClasses)}
            aria-describedby="result-detail-title"
          >
            {patient ? formatFullName(patient) : "--"}
          </span>
        </div>
        <div className="grid-col padding-2">
          <span
            className={classnames(
              labelClasses,
              "display-block margin-bottom-1"
            )}
          >
            Date of birth
          </span>
          <span
            className={classnames("font-sans-lg", removed && strikeClasses)}
            aria-describedby="result-detail-title"
          >
            {patient?.birthDate
              ? formatDateWithTimeOption(patient.birthDate)
              : "--"}
          </span>
        </div>
      </div>
      <h2 className="font-sans-md margin-top-3">Test information</h2>
      <table className={containerClasses}>
        <tbody>
          <DetailsRow
            label="COVID-19 result"
            value={displayResult["covidResult"]}
            removed={removed}
            aria-describedby="result-detail-title"
          />

          {multiplexEnabled ? (
            <>
              <DetailsRow
                label="Flu A result"
                value={displayResult["fluAResult"]}
                removed={removed}
                aria-describedby="result-detail-title"
              />
              <DetailsRow
                label="Flu B result"
                value={displayResult["fluBResult"]}
                removed={removed}
                aria-describedby="result-detail-title"
              />
            </>
          ) : (
            <></>
          )}
          <DetailsRow
            label="Test date"
            value={dateTested && formatDateWithTimeOption(dateTested, true)}
            removed={removed}
            aria-describedby="result-detail-title"
          />
          <DetailsRow
            label="Device"
            value={deviceType?.name}
            removed={removed}
            aria-describedby="result-detail-title"
          />
          <DetailsRow
            label="Symptoms"
            value={
              symptomList.length > 0 ? symptomList.join(", ") : "No symptoms"
            }
            removed={removed}
            aria-describedby="result-detail-title"
          />
          <DetailsRow
            label="Symptom onset"
            value={symptomOnset && formatDateWithTimeOption(symptomOnset)}
            removed={removed}
            aria-describedby="result-detail-title"
          />
          <DetailsRow
            label="Pregnant?"
            value={pregnancy && pregnancyMap[pregnancy]}
            removed={removed}
            aria-describedby="result-detail-title"
          />
          <DetailsRow
            label="Submitted by"
            value={createdBy?.name && formatFullName(createdBy.name)}
            removed={removed}
            last
            aria-describedby="result-detail-title"
          />
        </tbody>
      </table>
    </>
  );
};

const TestResultDetailsModal = (props: TestResultDetailsModalProps) => (
  <Modal
    isOpen={props.isOpen}
    style={{
      content: {
        maxHeight: "90vh",
        width: "50em",
        position: "initial",
      },
    }}
    overlayClassName="prime-modal-overlay display-flex flex-align-center flex-justify-center"
    contentLabel={`Result details for ${
      props.testResult?.patient
        ? displayFullName(
            props.testResult.patient.firstName,
            props.testResult.patient.middleName,
            props.testResult.patient.lastName
          )
        : "selected patient"
    }`}
    ariaHideApp={process.env.NODE_ENV !== "test"}
    onRequestClose={props.closeModal}
  >
    {props.testResult?.id && (
      <QueryWrapper<DetachedTestResultDetailsModalProps>
        query={testResultDetailsQuery}
        queryOptions={{ variables: { id: props.testResult.id } }}
        Component={DetachedTestResultDetailsModal}
        componentProps={{ ...props }}
        displayLoadingIndicator={false}
      />
    )}
  </Modal>
);

export default TestResultDetailsModal;

type StringOrFalsey = string | null | undefined | false;

const DetailsRow = ({
  label,
  value,
  indent,
  last,
  removed,
}: {
  label: string;
  value: StringOrFalsey;
  indent?: boolean;
  last?: boolean;
  removed?: boolean;
}) => {
  const tdClasses = classnames(
    "padding-y-3",
    !last && "border-bottom-1px border-base-lighter"
  );

  const labelColumnClasses = classnames(
    labelClasses,
    tdClasses,
    "width-card-lg text-left",
    indent && "font-sans-sm padding-left-2"
  );

  return (
    <tr>
      <th className={labelColumnClasses}>{label}</th>
      <td className={classnames(tdClasses, removed && strikeClasses)}>
        {value ?? "--"}
      </td>
    </tr>
  );
};
