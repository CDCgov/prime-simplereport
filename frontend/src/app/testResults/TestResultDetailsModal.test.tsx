import { render, screen } from "@testing-library/react";
import ReactDOM from "react-dom";

import { TestResult } from "../testQueue/QueueItem";

import { DetachedTestResultDetailsModal } from "./TestResultDetailsModal";

const testResult = {
  dateTested: "2022-01-28T17:56:48.143Z",
  result: "NEGATIVE" as TestResult,
  correctionStatus: null,
  noSymptoms: false,
  symptoms: '{"00000":"false"}',
  symptomOnset: "myOnset",
  pregnancy: null,
  deviceType: {
    name: "Fake device",
  },
  patient: {
    firstName: "First",
    middleName: "Middle",
    lastName: "Last",
    birthDate: "08/07/1990",
  },
  createdBy: {
    name: {
      firstName: "firstName",
      middleName: "middle",
      lastName: "last",
    },
  },
};

window.print = jest.fn();

describe("TestResultDetailsModal", () => {
  let component: any;

  beforeEach(() => {
    ReactDOM.createPortal = jest.fn((element, _node) => {
      return element;
    }) as any;

    component = render(
      <DetachedTestResultDetailsModal
        data={{ testResult }}
        testResultId="id"
        closeModal={() => {}}
      />
    );
  });

  it("should render the test date and test time", () => {
    expect(screen.getByText("01/28/2022 5:56pm")).toBeInTheDocument();
  });

  it("matches screenshot", () => {
    expect(component).toMatchSnapshot();
  });
});
