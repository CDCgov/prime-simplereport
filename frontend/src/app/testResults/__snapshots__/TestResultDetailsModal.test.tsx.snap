// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestResultDetailsModal matches screenshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="ReactModal__Body--open"
  >
    <div
      data-react-modal-body-trap=""
      style="position: absolute; opacity: 0;"
      tabindex="0"
    />
    <div>
      <div
        class="ReactModal__Overlay ReactModal__Overlay--after-open prime-modal-overlay display-flex flex-align-center flex-justify-center"
      >
        <div
          aria-label="Unsaved changes to current user"
          aria-modal="true"
          class="ReactModal__Content ReactModal__Content--after-open"
          role="dialog"
          style="position: initial; top: 40px; left: 40px; right: 40px; bottom: 40px; border: 1px solid #ccc; background: rgb(255, 255, 255); overflow: auto; border-radius: 4px; outline: none; padding: 20px; max-height: 90vh; width: 50em;"
          tabindex="-1"
        >
          <div
            class="display-flex flex-justify"
          >
            <h1
              class="font-heading-lg margin-top-05 margin-bottom-0"
            >
              Test details
            </h1>
            <div
              class="sr-time-of-test-buttons"
            >
              <button
                class="modal__close-button"
                style="cursor: pointer;"
              >
                <img
                  alt="Close"
                  class="modal__close-img"
                  src="close.svg"
                />
              </button>
            </div>
          </div>
          <div
            class="border-top border-base-lighter margin-x-neg-205 margin-top-1"
          />
          <h2
            class="font-sans-md margin-top-3"
          >
            Patient
          </h2>
          <div
            class="width-full font-sans-md add-list-reset border-base-lighter border-2px radius-md padding-x-2 padding-y-1 grid-row flex-row padding-0"
          >
            <div
              class="grid-col padding-2 border-right-1px border-base-lighter"
            >
              <span
                class="text-bold text-no-strike text-ink display-block margin-bottom-1"
              >
                Name
              </span>
              <span
                class="font-sans-lg"
              >
                First Middle Last
              </span>
            </div>
            <div
              class="grid-col padding-2"
            >
              <span
                class="text-bold text-no-strike text-ink display-block margin-bottom-1"
              >
                Date of birth
              </span>
              <span
                class="font-sans-lg"
              >
                08/07/1990
              </span>
            </div>
          </div>
          <h2
            class="font-sans-md margin-top-3"
          >
            Test details
          </h2>
          <table
            class="width-full font-sans-md add-list-reset border-base-lighter border-2px radius-md padding-x-2 padding-y-1"
          >
            <tbody>
              <tr>
                <th
                  class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
                >
                  SARS-CoV-2 result
                </th>
                <td
                  class="padding-y-3 border-bottom-1px border-base-lighter"
                >
                  NEGATIVE
                </td>
              </tr>
              <tr>
                <th
                  class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
                >
                  Test date
                </th>
                <td
                  class="padding-y-3 border-bottom-1px border-base-lighter"
                >
                  01/28/2022 5:56pm
                </td>
              </tr>
              <tr>
                <th
                  class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
                >
                  Device
                </th>
                <td
                  class="padding-y-3 border-bottom-1px border-base-lighter"
                >
                  Fake device
                </td>
              </tr>
              <tr>
                <th
                  class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
                >
                  Symptoms
                </th>
                <td
                  class="padding-y-3 border-bottom-1px border-base-lighter"
                >
                  No symptoms
                </td>
              </tr>
              <tr>
                <th
                  class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left font-sans-sm padding-left-2"
                >
                  Symptom onset
                </th>
                <td
                  class="padding-y-3 border-bottom-1px border-base-lighter"
                >
                  Invalid date
                </td>
              </tr>
              <tr>
                <th
                  class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
                >
                  Pregnant?
                </th>
                <td
                  class="padding-y-3 border-bottom-1px border-base-lighter"
                >
                  --
                </td>
              </tr>
              <tr>
                <th
                  class="text-bold text-no-strike text-ink padding-y-3 width-card-lg text-left"
                >
                  Submitted by
                </th>
                <td
                  class="padding-y-3"
                >
                  firstName middle last
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
    <div
      data-react-modal-body-trap=""
      style="position: absolute; opacity: 0;"
      tabindex="0"
    />
    <div
      class="ReactModalPortal"
    />
  </body>,
  "container": <div>
    <div
      class="ReactModal__Overlay ReactModal__Overlay--after-open prime-modal-overlay display-flex flex-align-center flex-justify-center"
    >
      <div
        aria-label="Unsaved changes to current user"
        aria-modal="true"
        class="ReactModal__Content ReactModal__Content--after-open"
        role="dialog"
        style="position: initial; top: 40px; left: 40px; right: 40px; bottom: 40px; border: 1px solid #ccc; background: rgb(255, 255, 255); overflow: auto; border-radius: 4px; outline: none; padding: 20px; max-height: 90vh; width: 50em;"
        tabindex="-1"
      >
        <div
          class="display-flex flex-justify"
        >
          <h1
            class="font-heading-lg margin-top-05 margin-bottom-0"
          >
            Test details
          </h1>
          <div
            class="sr-time-of-test-buttons"
          >
            <button
              class="modal__close-button"
              style="cursor: pointer;"
            >
              <img
                alt="Close"
                class="modal__close-img"
                src="close.svg"
              />
            </button>
          </div>
        </div>
        <div
          class="border-top border-base-lighter margin-x-neg-205 margin-top-1"
        />
        <h2
          class="font-sans-md margin-top-3"
        >
          Patient
        </h2>
        <div
          class="width-full font-sans-md add-list-reset border-base-lighter border-2px radius-md padding-x-2 padding-y-1 grid-row flex-row padding-0"
        >
          <div
            class="grid-col padding-2 border-right-1px border-base-lighter"
          >
            <span
              class="text-bold text-no-strike text-ink display-block margin-bottom-1"
            >
              Name
            </span>
            <span
              class="font-sans-lg"
            >
              First Middle Last
            </span>
          </div>
          <div
            class="grid-col padding-2"
          >
            <span
              class="text-bold text-no-strike text-ink display-block margin-bottom-1"
            >
              Date of birth
            </span>
            <span
              class="font-sans-lg"
            >
              08/07/1990
            </span>
          </div>
        </div>
        <h2
          class="font-sans-md margin-top-3"
        >
          Test details
        </h2>
        <table
          class="width-full font-sans-md add-list-reset border-base-lighter border-2px radius-md padding-x-2 padding-y-1"
        >
          <tbody>
            <tr>
              <th
                class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
              >
                SARS-CoV-2 result
              </th>
              <td
                class="padding-y-3 border-bottom-1px border-base-lighter"
              >
                NEGATIVE
              </td>
            </tr>
            <tr>
              <th
                class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
              >
                Test date
              </th>
              <td
                class="padding-y-3 border-bottom-1px border-base-lighter"
              >
                01/28/2022 5:56pm
              </td>
            </tr>
            <tr>
              <th
                class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
              >
                Device
              </th>
              <td
                class="padding-y-3 border-bottom-1px border-base-lighter"
              >
                Fake device
              </td>
            </tr>
            <tr>
              <th
                class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
              >
                Symptoms
              </th>
              <td
                class="padding-y-3 border-bottom-1px border-base-lighter"
              >
                No symptoms
              </td>
            </tr>
            <tr>
              <th
                class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left font-sans-sm padding-left-2"
              >
                Symptom onset
              </th>
              <td
                class="padding-y-3 border-bottom-1px border-base-lighter"
              >
                Invalid date
              </td>
            </tr>
            <tr>
              <th
                class="text-bold text-no-strike text-ink padding-y-3 border-bottom-1px border-base-lighter width-card-lg text-left"
              >
                Pregnant?
              </th>
              <td
                class="padding-y-3 border-bottom-1px border-base-lighter"
              >
                --
              </td>
            </tr>
            <tr>
              <th
                class="text-bold text-no-strike text-ink padding-y-3 width-card-lg text-left"
              >
                Submitted by
              </th>
              <td
                class="padding-y-3"
              >
                firstName middle last
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
